dio/lib/src/adapters/browser_adapter.dart

```diff
@@ -178,6 +178,24 @@ class BrowserHttpClientAdapter implements HttpClientAdapter {
       );
     });
 
+    xhr.onTimeout.first.then((_) {
+      int timeoutMilliseconds = 0;
+      if (connectTimeout != null) {
+        timeoutMilliseconds += connectTimeout.inMilliseconds;
+      }
+      if (receiveTimeout != null) {
+        timeoutMilliseconds += receiveTimeout.inMilliseconds;
+      }
+      connectTimeoutTimer?.cancel();
+      completer.completeError(
```

Comment:
One thing I've missed is we should check if the completer is completed before calling this.

Reply:
@kuhnroyal Added :) Thanks!

Reply:
This seems still not addressed.

Reply:
@AlexV525 I think the preview here still displays the old code, go to the "Files changed" section, you'll see on the latest code version:
```
      if (connectTimeoutTimer != null) {
        connectTimeoutTimer?.cancel();
      }
```

Reply:
@AlexV525 Sorry my bad, misunderstood what you meant, now I think I handled it properly :)